ACViewer-SDK/Assets/SkinsModule/SkinController.cs

```diff
@@ -0,0 +1,304 @@
+    }
+
+
+    void FixedUpdate()
+    {
+        permissionTimer += Time.fixedDeltaTime;
+#if PLATFORM_ANDROID || UNITY_IOS
+        if (Permission.HasUserAuthorizedPermission(Permission.FineLocation) && !geoStarted)
+        {
+            StartCoroutine(GetUIParams(UIParamsParcer));
+            geoStarted = true;
+        }
+#endif
+        if (!geoStarted && permissionTimer > permissionTime) 
+        {
+            panelNoGeoPermisson.SetActive(true);
+            geoStarted = true;
+        }
+    }
+
+    IEnumerator CheckLocationS()
+    {
+        if (!Input.location.isEnabledByUser)
+        {
+            Debug.Log("geo not enabled");
+        }
+        // Start service before querying location
+        Input.location.Start();
+        // Wait until service initializes
+        int maxWait = 4;
+        while (Input.location.status == LocationServiceStatus.Initializing && maxWait > 0)
+        {
+            yield return new WaitForSeconds(0.5f);
+            maxWait--;
+        }
+
+        // Service didn't initialize in 20 seconds
+        if (maxWait < 1)
+        {
+            Debug.Log("Timed out");
+            yield break;
+        }
+
+        // Connection has failed
+        if (Input.location.status == LocationServiceStatus.Failed)
+        {
+            Debug.Log("Unable to determine device location");
+            yield break;
+        }
+        else
+        {
+            // Access granted and location value could be retrieved
+            Debug.Log("Location: " + Input.location.lastData.latitude + " " + Input.location.lastData.longitude + " " + Input.location.lastData.altitude + " " + Input.location.lastData.horizontalAccuracy + " " + Input.location.lastData.timestamp);
+            lat = Input.location.lastData.latitude;
+            lon = Input.location.lastData.longitude;
+        }
+    }
+
+    public double DistanceTo(double lat1, double lon1, double lat2, double lon2, char unit = 'K')
+    {
+        double rlat1 = Math.PI * lat1 / 180;
+        double rlat2 = Math.PI * lat2 / 180;
+        double theta = lon1 - lon2;
+        double rtheta = Math.PI * theta / 180;
+        double dist =
+            Math.Sin(rlat1) * Math.Sin(rlat2) + Math.Cos(rlat1) *
+            Math.Cos(rlat2) * Math.Cos(rtheta);
+        dist = Math.Acos(dist);
+        dist = dist * 180 / Math.PI;
+        dist = dist * 60 * 1.1515;
+        switch (unit)
+        {
+            case 'K': //Kilometers -> default
+                return dist * 1.609344;
+            case 'N': //Nautical Miles 
+                return dist * 0.8684;
+            case 'M': //Miles
+                return dist;
+        }
+        return dist;
+    }
+
+    IEnumerator GetUIParams(Action<string> getJsonParams)
+    {
+#if !UNITY_EDITOR
+            yield return StartCoroutine(CheckLocationS());
+#endif
+        //var sw = UnityWebRequest.Get(apiURL + "/media/resources/ACV/molfetta.props_by_gps.json"); //Molfetta params
```

Comment:
Kill it, or simplify the comment - keep only prefix "molfetta." to the filename.